src/runtime/nitro/renderer.ts

```diff
@@ -90,7 +90,7 @@ export default eventHandler(async (event) => {
     event,
     req: event.req,
     res: event.res,
-    runtimeConfig: { private: config, public: { public: config.public, app: config.app } },
+    runtimeConfig: { private: config, public: { ...config.public, public: config.public, app: config.app } },
```

Comment:
It should be handled by the runtime proxy.

Reply:
As i understand runtime proxy is a config object itself that comes from composable useRuntimeConfig , but here its set values as a new object with public and app path, which  nullify that proxy so it leads to no props at root like nuxt 2 expect